### PR TITLE
Add option to not use estimator.

### PR DESCRIPTION
**Pull request by reedwm**
To test convergence, I ran the following 48 times.
```
python ncf_main.py --model_dir=$HOME/ncf_model_dir_gpu_0 --data_dir=$HOME/ncf_data_dir_gpu_0 --dataset ml-20m --hooks= --num_gpus -1 --clean --train_epochs 14 --batch_size 65536 --eval_batch_size 65536 --layers 256,256,128,64 --num_factors 64 --hr_threshold 0.99 --ml_perf --learning_rate=0.00281936 --beta1=0.0490852 --beta2=0.965095 --epsilon=1.23757e-08 --use_xla_for_gpu --nouse_estimator
```

I hit a hit-rate of 63.5 44/48 times.

### CODE REVIEW

File: official/recommendation/model_runner.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      initializer_feed_dict = None
+      record_dir = None
+
+    self._session.run(model_properties.iterator.initializer,
+                      initializer_feed_dict)
+    fetches = (model_properties.loss, model_properties.run_model_op)
+    mode = "Train" if is_training else "Eval"
+    start = None
+    for i in range(num_steps):
+      loss, _, = self._session.run(fetches)
```
Comment (robieta) [inline official/recommendation/model_runner.py]: You've tested that there is no overhead from calling this one at a time?

File: official/recommendation/ncf_main.py [hunk 2]
```diff
@@ -237,37 +244,46 @@ def run_ncf(_):
...
-    if train_record_dir:
-      tf.gfile.DeleteRecursively(train_record_dir)
-
-    tf.logging.info("Beginning evaluation.")
-    if pred_input_fn is None:
-      pred_input_fn, _, eval_batch_count = data_preprocessing.make_input_fn(
-          ncf_dataset=ncf_dataset, is_training=False)
-
-      if eval_batch_count != num_eval_steps:
+    if FLAGS.use_estimator:
```
Comment (robieta) [inline official/recommendation/ncf_main.py]: Have you confirmed that the estimator path still works? (1 run is sufficient.)

File: official/recommendation/model_runner.py (hunk 1 above)
Comment (reedwm) [reply official/recommendation/model_runner.py]: I haven't measured, but this is still better than what Estimator does. If there is overhead, we can further optimize this.

File: official/recommendation/ncf_main.py (hunk 2 above)
Comment (reedwm) [reply official/recommendation/ncf_main.py]: I ran for two epochs on ml-1m, which worked.

File: official/recommendation/model_runner.py (hunk 1 above)
Comment (robieta) [reply official/recommendation/model_runner.py]: Sounds good.

File: official/recommendation/ncf_main.py (hunk 2 above)
Comment (robieta) [reply official/recommendation/ncf_main.py]: good 'nuf.

**Pull request merged**